src/mono/wasm/debugger/tests/debugger-test/debugger-test.cs

```diff
@@ -518,3 +518,15 @@ public static void run()
         StaticMethodWithLocalEmptyStructAsync().Wait();
     }
 }
+
+public class LoadDebuggerTest {
+    public static void LoadLazyAssembly(string asm_base64, string pdb_base64)
+    {
+        byte[] asm_bytes = Convert.FromBase64String(asm_base64);
+        byte[] pdb_bytes = Convert.FromBase64String(pdb_base64);
+
+        var loadedAssembly = System.Runtime.Loader.AssemblyLoadContext.Default.LoadFromStream(new System.IO.MemoryStream(asm_bytes), new System.IO.MemoryStream(pdb_bytes));
```

Comment:
It works, I have already tested, and it's a good idea to add it in the tests.

Reply:
> Is that a valid case for us?

At least for the Blazor scenario, we'll always load into the default context so there's no valid case there.

> It works, I have already tested, and it's a good idea to add it in the tests.

Sure. I can add a test.

Reply:
So, this works-ish. When loaded into a separate context, the callback is triggered, PDBs are loaded into the DebugStore, and scripts are parsed.

However, setting breakpoints fails because we currently do a lookup for the assembly in the default context ([ref](https://github.com/dotnet/runtime/blob/master/src/mono/mono/mini/mini-wasm-debugger.c#L513)).

Not sure if we want to address this now considering we don't anticipate loading into the non-default context for the Blazor scenario.

I imagine supporting this will require modifying the `mono_wasm_set_breakpoint` method to support looks up in multiple contexts.

Reply:
@captainsafia (or Ankit or Thays) could you create an issue for MonoVM to support for setting breakpoints in non-default ALCs?

I think it probably affects the non-wasm scenarios too.  Actually I wonder what happens if we load the same assembly into multiple ALCs

Reply:
I can create the issue.